core-bundle/src/Image/Studio/TemplateData.php

```diff
@@ -0,0 +1,311 @@
+                'imgSize' => sprintf(' width="%d" height="%d"', $originalSize->getWidth(), $originalSize->getHeight()),
+                'singleSRC' => $this->studio->getFilePath(),
+                'fullsize' => \in_array($this->linkMode, [self::LINK_NEW_WINDOW, self::LINK_LIGHTBOX], true),
+                'margin' => $this->marginProperty ?? '',
+                'addBefore' => 'below' !== $this->floatingProperty,
+                'addImage' => true,
+            ]
+        );
+
+        // Context sensitive properties
+        if (null !== ($title = $metaData->getTitle())) {
+            $templateData['picture']['title'] = $title;
+        }
+
+        if (null !== $this->floatingProperty) {
+            $templateData['floatClass'] = " float_{$this->floatingProperty}";
+        }
+
+        // Link and lightbox attributes
+        if (!empty($attributes = $this->getAttributes())) {
+            $htmlAttributes = array_map(
+                static function (string $attribute, string $value) {
+                    return sprintf('%s="%s"', $attribute, $value);
+                },
+                array_keys($attributes), $attributes
+            );
+
+            $templateData['attributes'] = ' '.implode(' ', $htmlAttributes);
+        }
+
+        // Link target
+        if (null !== ($href = $this->getHref())) {
+            $templateData['href'] = $href;
+        }
+
+        // Secondary image
+        if (null !== $this->secondaryStudio) {
+            if (!empty($templateData['imageTitle']) && empty($templateData['linkTitle'])) {
+                $templateData['linkTitle'] = $templateData['imageTitle'];
+                unset($templateData['imageTitle']);
+            }
+
+            $templateData['lightboxPicture'] = [
+                'img' => $this->secondaryStudio->getImg(),
+                'sources' => $this->secondaryStudio->getSources(),
+            ];
+        }
+
+        return $templateData;
+    }
+
+    /**
+     * Apply the template data to an existing template. This handles special
+     * cases to prevent overriding existing keys / values.
+     */
+    public function applyToTemplate(Template $template): void
+    {
+        $new = $this->getData();
+        $existing = $template->getData();
+
+        // Do not override the "href" key (see #6468)
+        if (isset($new['href'], $existing['href'])) {
+            $new['imageHref'] = $new['href'];
+            unset($new['href']);
+        }
+
+        // Append attributes instead of replacing
+        if (isset($new['attributes'], $existing['attributes'])) {
+            $new['attributes'] = ($existing['attributes'] ?? '').$new['attributes'];
+        }
+
+        $template->setData(array_replace_recursive($existing, $new));
+    }
+
+    private function getFallbackLightBoxId(): string
+    {
+        // Try to generate a unique identifier that is stable across calls
+        $identifier = $this->studio->getMetaData()->getUrl() ?? $this->studio->getFilePath();
```

Comment:
We could use the file path of the secondary studio if it exists I think.

Reply:
You mean effectively swapping the arguments?

 ```php
$identifier = $this->studio->getFilePath() ?? $this->studio->getMetaData()->getUrl();
```

Reply:
more something like that: 
```suggestion
        $identifier = $this->secondaryStudio->getFilePath() ?? $this->studio->getMetaData()->getUrl() ?? $this->studio->getFilePath();
```

Reply:
Ah got it. Makes sense.